osgar/drivers/eduro.py

```diff
@@ -0,0 +1,228 @@
+"""
+  Driver for robot Eduro
+"""
+
+
+import ctypes
+import serial
+import struct
+import math
+from threading import Thread
+
+from osgar.logger import LogWriter, LogReader
+from osgar.bus import BusShutdownException
+from .canserial import CAN_packet
+
+CAN_ID_SYNC = 0x80
+CAN_ID_ENCODERS_LEFT = 0x181
+CAN_ID_ENCODERS_RIGHT = 0x182
+CAM_SYSTEM_STATUS = 0x8A  # including emergency STOP button
+CAN_ID_BUTTONS = 0x28A
+
+UPDATE_TIME_FREQUENCY = 20.0  # Hz
+WHEEL_DIAMETER_LEFT = 427.0 / 445.0 * 0.26/4.0 + 0.00015
+WHEEL_DIAMETER_RIGHT = 427.0 / 445.0 * 0.26/4.0 - 0.00015
+
+
+def sint32_diff(a, b):
+    return ctypes.c_int32(a - b).value
+
+
+class Eduro(Thread):
+    def __init__(self, config, bus):
+        Thread.__init__(self)
+        self.setDaemon(True)
+
+        self.bus = bus
+
+        self.desired_speed = 0.0  # m/s
+        self.desired_angular_speed = 0.0
+        self.maxAcc = config.get('max_acc', 0.5)
+        self._rampLastLeft, self._rampLastRight = 0.0, 0.0
+        self.WHEEL_DISTANCE = 0.315        
```

Comment:
Why is WHEEL_DISTANCE in the class but others are in the module? Should this also by in the module? Can this change?

Reply:
OK, good point. There used to be "small Eduro" and "Eduro Maxi", but there is probably no plan to return back to small Eduro -> fixed.